### PR TITLE
[Holy Paladin] Vision of Perfection Essence

### PR DESCRIPTION
**Pull request by 5chmidt**
Another item for Glimmer Build #3409, 
- Added major statistic box to show users how often their Vision of Perfection traits would proc.  
- Added damage and healing stats for major essence.
- Added minor statistic box showing versatility and healing 
- Added haste and versatility buffs to the stat tracker.
- Added a pre-pull check for Avenging Crusader.
- Added a cooldown reduction for Avenging Crusader if Strive for Perfection is used.

image

[Example log](https://www.warcraftlogs.com/reports/NXkDdR1LnhAcPabq#fight=12&type=healing&source=3)

### CODE REVIEW

File: src/parser/paladin/holy/modules/Buffs.js [hunk 1]
```diff
@@ -48,6 +48,10 @@ class Buffs extends CoreBuffs {
         enabled: combatant.hasTalent(SPELLS.HOLY_AVENGER_TALENT),
         timelineHightlight: true,
       },
+      {
+        spellId: SPELLS.AVENGING_WRATH.id,
```
Comment (MartijnHols) [inline src/parser/paladin/holy/modules/Buffs.js]: This spell, as well as AC, already exist in the buff file. Shouldn't those be updated instead?

File: src/parser/paladin/holy/modules/Abilities.tsx [hunk 2]
```diff
@@ -184,7 +184,7 @@ class Abilities extends CoreAbilities {
       {
         spell: SPELLS.AVENGING_WRATH,
         category: Abilities.SPELL_CATEGORIES.COOLDOWNS,
-        cooldown: 120,
+        cooldown: (this.selectedCombatant.hasMajor(SPELLS.VISION_OF_PERFECTION.traitId)) ? 120 * 0.79 : 120,
```
Comment (MartijnHols) [inline src/parser/paladin/holy/modules/Abilities.tsx]: How about AC?

File: src/parser/paladin/holy/modules/Buffs.js (hunk 1 above)
Comment (5chmidt) [reply src/parser/paladin/holy/modules/Buffs.js]: Yea, you are correct.  This is fixed in the latest PR.

File: src/parser/paladin/holy/modules/Abilities.tsx (hunk 2 above)
Comment (5chmidt) [reply src/parser/paladin/holy/modules/Abilities.tsx]: Yes, both flavors of wings cooldowns are effected.  I have updated them accordingly.

File: src/parser/paladin/holy/modules/azeritetraits/GlimmerOfLight.js [hunk 3]
```diff
@@ -157,7 +157,7 @@ class GlimmerOfLight extends Analyzer {
           </>
         )}
         tooltip={(
-          <Trans>
+          <>
```
Comment (5chmidt) [inline src/parser/paladin/holy/modules/azeritetraits/GlimmerOfLight.js]: I am not sure why this is.  The <Trans> tag works fine when I build locally but when it is deployed it does not complete the string formatting.

File: src/parser/core/Combatant.ts [hunk 4]
```diff
@@ -180,7 +180,7 @@ class Combatant extends Entity {
     return Boolean(this.essencesByTraitID[traitId]);
   }
   hasMajor(traitId: number) {
-    return this.essencesByTraitID[traitId].isMajor;
+    return this.essencesByTraitID[traitId] && this.essencesByTraitID[traitId].isMajor;
```
Comment (5chmidt) [inline src/parser/core/Combatant.ts]: I must have missed this earlier, as I thought previous tests were passing.

**Pull request merged**